test/extensions/filters/http/cache/cache_filter_test.cc

```diff
@@ -5,122 +5,255 @@
 #include "source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.h"
 
 #include "test/extensions/filters/http/cache/common.h"
+#include "test/extensions/filters/http/cache/mocks.h"
 #include "test/mocks/server/factory_context.h"
 #include "test/test_common/simulated_time_system.h"
 #include "test/test_common/utility.h"
 
 #include "gtest/gtest.h"
 
+using ::testing::AllOf;
+using ::testing::ByMove;
+using ::testing::Field;
+using ::testing::InvokeArgument;
+using ::testing::InvokeWithoutArgs;
+using ::testing::Ref;
+using ::testing::ReturnRef;
+using ::testing::StrictMock;
+
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace Cache {
 namespace {
 
-class CacheFilterTest : public ::testing::Test {
-protected:
-  // The filter has to be created as a shared_ptr to enable shared_from_this() which is used in the
-  // cache callbacks.
-  CacheFilterSharedPtr makeFilter(HttpCache& cache) {
-    auto filter = std::make_shared<CacheFilter>(config_, /*stats_prefix=*/"", context_.scope(),
-                                                context_.timeSource(), cache);
-    filter->setDecoderFilterCallbacks(decoder_callbacks_);
-    filter->setEncoderFilterCallbacks(encoder_callbacks_);
-    return filter;
+constexpr absl::string_view kBody = "abc";
+
+// Returns a std::function that can be used as an action with
+// MockLookupContext::getBody to return the provided body or chunk.
+std::function<void(const AdjustedByteRange&, LookupBodyCallback&&)>
+getBodyChunk(absl::string_view chunk) {
+  return [chunk = std::string(chunk)](const AdjustedByteRange&, LookupBodyCallback&& cb) {
+    cb(std::make_unique<Envoy::Buffer::OwnedImpl>(chunk));
```

Comment:
It probably doesn't matter cause it's just in tests, but will this copy the chunk twice, once for the function and once for the unique_ptr constructor?

Reply:
Yeah. It's hard to avoid that without doing some backflips because gmock needs actions to be copiable, so for the test I figured it wasn't worth it.